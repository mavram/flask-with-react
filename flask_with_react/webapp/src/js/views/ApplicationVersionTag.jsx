
import React from 'react'
import {withService, Service} from '../modules/Service'
import {ApplicationService} from '../Services'


const ApplicationVersionTagView = ({service, status, data}) => {
    if (status === Service.INITIALIZING) {
        return <p className='lead' onClick = {() => service.get()}> Application version not loaded. Click to load </p>
    } else if (status === Service.FAILED) {
        return <p className='lead'>Failed to load application version. {data.message}</p>
    } else if (status === Service.LOADING) {
        return <p className='lead'>Loading application version...</p>
    }
    return <p className='lead'>Application version: {data.version}</p>
}


export const ApplicationVersionTag = withService(ApplicationVersionTagView, ApplicationService)
