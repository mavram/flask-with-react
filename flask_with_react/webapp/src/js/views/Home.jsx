
import React from 'react'


export const Home = () => {
    return <div className='container'>
        <div className='row'>
            <div className='col'>
                <p className='lead'>Home page. Authenticated.</p>
            </div>
        </div>
    </div>
}
