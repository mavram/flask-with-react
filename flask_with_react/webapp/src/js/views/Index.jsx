
import React from 'react'
import {Redirect} from 'react-router-dom'
import {Settings, JWT_KEY} from '../Settings'
import {withService, Service} from '../modules/Service'
import {LoginService} from '../Services'


const LoginView = ({status, service, data}) => {
    if (status === Service.DONE) {
        Settings.set(JWT_KEY, data)
        return <Redirect to='/home'/>
    }

    const getMessage = () => {
        if (status === Service.LOADING) {
            return 'Authenticating...'
        } else if (status === Service.FAILED) {
            return 'Failed to authenticate.'
        } else if (status === Service.INITIALIZING) {
            return 'Initializing...'
        }
        return 'N/A'
    }

    return <div className='container'>
        <div className='row'>
            <div className='col'>
                <p className='lead'>{getMessage()}</p>
            </div>
        </div>
        <div className='row'>
            <div className='col'>
                <button onClick={() => service.login('admin', 'passwd')}>Login</button>
            </div>
        </div>
    </div>
}


const LoginWithService = withService(LoginView, LoginService)


export const Login = () => {
    return <LoginWithService/>
}


export const Logout = () => {
    if (Settings.isAuthenticated()) {
        Settings.set(JWT_KEY)
    }
    return <Redirect to='/'/>
}


export const Index = () => {
    if (Settings.isAuthenticated()) {
        return <Redirect to='/home'/>
    }
    return <Redirect to='/welcome'/>
}
