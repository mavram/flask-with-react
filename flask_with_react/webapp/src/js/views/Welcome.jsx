
import React from 'react'
import {Link} from 'react-router-dom'


export const Welcome = () => {
    return <div className='container'>
        <div className='row'>
            <div className='col'>
                <p className='lead'>Welcome page. Not-Authenticated.</p>
                <p className='lead'>
                    <Link className='no-hover-decoration' to='/login'>Login...</Link>
                </p>
            </div>
        </div>
    </div>
}
