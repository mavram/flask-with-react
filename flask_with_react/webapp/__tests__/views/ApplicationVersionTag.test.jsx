import React from 'react'
import renderer from 'react-test-renderer'
import {ApplicationVersionTag} from '../../src/js/views/ApplicationVersionTag'


it('ApplicationVersionTag basic creation', () => {
    const component = renderer.create(
        <ApplicationVersionTag />
    )

    expect(component.toJSON()).toMatchSnapshot()
})
