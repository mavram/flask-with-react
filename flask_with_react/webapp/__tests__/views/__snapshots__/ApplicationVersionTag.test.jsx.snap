// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationVersionTag basic creation 1`] = `
<p
  className="lead"
  onClick={[Function]}
>
   Application version not loaded. Click to load 
</p>
`;
